docs/vc-spec.md

```diff
@@ -11,60 +11,82 @@ The Candid interface is as follows, and the subsequent sections describe the
 services and the corresponding messages in more detail.
 
 ```candid
-type ArgumentValue = variant { "Int" : int32; String : text };
+// Specification of a requested credential.
 type CredentialSpec = record {
-    arguments : opt vec record { text; ArgumentValue };
     credential_type : text;
+    /// arguments are optional, and specific to the credential_type
+    arguments : opt vec record { text; ArgumentValue };
 };
-type GetCredentialRequest = record {
-    signed_id_alias : SignedIdAlias;
-    prepared_context : opt blob;
-    credential_spec : CredentialSpec;
-};
-type GetCredentialResponse = variant {
-    Ok : IssuedCredentialData;
-    Err : IssueCredentialError;
-};
+type ArgumentValue = variant { "Int" : int32; String : text };
+
+/// Messages for ICRC-21 consent message, cf.
+/// https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_21_consent_msg.md
 type Icrc21ConsentInfo = record { consent_message : text; language : text };
-type Icrc21VcConsentMessageRequest = record {
-    preferences : Icrc21ConsentPreferences;
-    credential_spec : CredentialSpec;
-};
-type Icrc21ConsentMessageResponse = variant {
-    Ok : Icrc21ConsentInfo;
-    Err : Icrc21Error;
-};
 type Icrc21ConsentPreferences = record { language : text };
 type Icrc21Error = variant {
-    GenericError : Icrc21ErrorInfo;
+    GenericError : record { description : text; error_code : nat };
```

Comment:
There is also `InsufficientPayment` now (if you are already updating these types).

Reply:
I'll add it in a yet another PR, as this requires a change of code as well.